### PR TITLE
admin crud category

### PR DESCRIPTION
**Pull request by hoangdv99**
## Related Tickets:
- [#40592](https://edu-redmine.sun-asterisk.vn/issues/40592)

## WHAT this PR does?

Admin crud category
## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Review by bachtung144 — APPROVED** [review]

**Comment by hoangdv99** [comment]
ready

File: app/Http/Livewire/Admin/AdminAddCategoryComponent.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public function generateSlug()
+    {
+        $this->slug = Str::slug($this->name);
+    }
+
+    public function storeCategory(Request $request)
+    {
+        $this->validate();
```
Comment (huyenttm-0901) [inline app/Http/Livewire/Admin/AdminAddCategoryComponent.php:36]: dùng formrequest đi em

File: app/Http/Livewire/Admin/AdminEditCategoryComponent.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public $categoryId;
+    public $name;
+    public $slug;
+
+    public function mount($categorySlug)
+    {
+        $this->categorySlug = $categorySlug;
+        $category = Category::where('slug', $categorySlug)->first();
+        $this->categoryId = $category->id;
+        $this->name = $category->name;
```
Comment (huyenttm-0901) [inline app/Http/Livewire/Admin/AdminEditCategoryComponent.php:21]: em có validate data chưa?

**Comment by hoangdv99** [comment]
ready

File: app/Http/Livewire/Admin/AdminAddCategoryComponent.php (hunk 1 above)
Comment (hoangdv99) [reply app/Http/Livewire/Admin/AdminAddCategoryComponent.php:36]: livewire ko dùng đc form request

File: app/Http/Livewire/Admin/AdminEditCategoryComponent.php (hunk 2 above)
Comment (hoangdv99) [reply app/Http/Livewire/Admin/AdminEditCategoryComponent.php:42]: đã thêm

**Review by manhtd-1083 — APPROVED** [review]

**Pull request merged**